nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -416,7 +420,8 @@ in {
     config = {
       dbtype = mkOption {
         type = types.enum [ "sqlite" "pgsql" "mysql" ];
-        default = "sqlite";
+        default = if versionAtLeast "25.05" config.system.stateVersion then "pgsql" else "sqlite";
```

Comment:
does it make sense to enable `createLocally` too when defaulting for postgres?

Reply:
This would silently activate a postgresql for everyone who has nextcloud already configured, but with a remote database.
So I'd prefer to keep this opt-in.

Reply:
i would tie it to the stateVersion also, but i am also okay in keeping it as it is.

Reply:
Now that the defaults are dropped anyways, I think this can be resolved.